src/Analyser/TypeSpecifier.php

```diff
@@ -709,18 +710,48 @@ public function specifyTypesInCondition(
 
 				$isset = $scope->issetCheck($issetExpr, static fn () => true);
 
-				if ($isset !== true) {
+				if ($isset === false) {
 					return new SpecifiedTypes();
 				}
 
-				return $this->create(
+				$exprType = $this->create(
 					$issetExpr,
 					new NullType(),
 					$context->negate(),
 					false,
 					$scope,
 					$rootExpr,
 				);
+				if ($isset === true) {
+					return $exprType;
+				}
+
+				if ($issetExpr instanceof Expr\Variable && is_string($issetExpr->name)) {
```

Comment:
hopefully its OK to do the variable stuff in a separate PR.
whats missing is `ArrayDimFetch`, `PropertyFetch`, `StaticPropertyFetch` which could be done in a followup?